gateway-service-impl/src/main/java/org/hypertrace/gateway/service/baseline/BaselineServiceQueryParser.java

```diff
@@ -0,0 +1,229 @@
+      long startTimeInMillis,
+      long endTimeInMillis,
+      List<String> entityIdAttributes,
+      String timeColumn) {
+    // adds the Id != "null" filter to remove null entities.
+    Filter.Builder filterBuilder =
+        Filter.newBuilder()
+            .setOperator(Operator.AND)
+            .addAllChildFilter(
+                entityIdAttributes.stream()
+                    .map(
+                        entityIdAttribute ->
+                            QueryRequestUtil.createColumnValueFilter(
+                                entityIdAttribute, Operator.NEQ, QUERY_SERVICE_NULL))
+                    .map(Filter.Builder::build)
+                    .collect(Collectors.toList()));
+
+    // Time range is a mandatory filter for query service, hence add it if it's not already present.
+    filterBuilder.addChildFilter(
+        Filter.newBuilder()
+            .setOperator(Operator.AND)
+            .addChildFilter(createTimeFilter(timeColumn, Operator.GE, startTimeInMillis))
+            .addChildFilter(createTimeFilter(timeColumn, Operator.LT, endTimeInMillis)));
+
+    return filterBuilder;
+  }
+
+  private static Filter createTimeFilter(String columnName, Operator op, long value) {
+    ColumnIdentifier.Builder timeColumn = ColumnIdentifier.newBuilder().setColumnName(columnName);
+    org.hypertrace.core.query.service.api.Expression.Builder lhs =
+        org.hypertrace.core.query.service.api.Expression.newBuilder()
+            .setColumnIdentifier(timeColumn);
+    LiteralConstant.Builder constant =
+        LiteralConstant.newBuilder()
+            .setValue(
+                org.hypertrace.core.query.service.api.Value.newBuilder()
+                    .setValueType(org.hypertrace.core.query.service.api.ValueType.LONG)
+                    .setLong(value));
+    org.hypertrace.core.query.service.api.Expression.Builder rhs =
+        org.hypertrace.core.query.service.api.Expression.newBuilder().setLiteral(constant);
+    return Filter.newBuilder().setLhs(lhs).setOperator(op).setRhs(rhs).build();
+  }
+
+  public BaselineEntitiesResponse parseQueryResponse(
+      Iterator<ResultSetChunk> resultSetChunkIterator,
+      BaselineRequestContext requestContext,
+      int idColumnsSize,
+      String entityType,
+      long periodSecs) {
+    Map<String, AttributeMetadata> attributeMetadataMap =
+        attributeMetadataProvider.getAttributesMetadata(requestContext, entityType);
+    Map<EntityKey, Map<String, BaselineMetricSeries.Builder>> entityMetricSeriesMap =
+        new LinkedHashMap<>();
+    while (resultSetChunkIterator.hasNext()) {
+      ResultSetChunk chunk = resultSetChunkIterator.next();
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Received chunk: " + chunk.toString());
+      }
+
+      if (chunk.getRowCount() < 1) {
+        break;
+      }
+
+      for (Row row : chunk.getRowList()) {
+        EntityKey entityKey =
+            EntityKey.of(
+                IntStream.range(0, idColumnsSize)
```

Comment:
This is the assumption taken across all the places in gateway service, this is copy pasted code from QueryServiceEntityFetcher. We should definitely refactor it and do it across all the places. Will raise a jira for that, but don't want to add to this since it is already big change.

Reply:
ok